### ISSUE TITLE
$tooltipProvider - how do I actually use setTriggers() ?

### ISSUE THREAD

**Issue by franciscopreller** [angular-ui/bootstrap#625]
The documentation states that I am able to set new triggers using the $tooltipProvider and setTriggers() method, however I have so far been unable to achieve this.

I have included this in my module config:

$tooltipProvider.setTriggers({ 'inputInvalid' : 'inputValid' });

However it does not find the method. I dived into the source and see the method there, but it's tossed inside a $get method which then returns the dependencies. Perhaps I am doing something wrong here and I simply do not understand how to get the $tooltipProvider to work?

**Comment by seadour** [comment]
@courtneyfaulkner Thank you so much for the Plunkr demo, very helpful to us Angular n00bs

**Comment by VictorCoding** [comment]
@courtneyfaulkner thanks alot! great plnkr. In my case i had to trigger the popover to close when selecting an item inside the popover.

**Comment by thotty** [comment]
Thx, @courtneyfaulkner.
:smile:

**Comment by courtneyfaulkner** [comment]
I'm happy I could help. :)

**Comment by cyrilgandon** [comment]
The plunler is broken with the 0.13.4...

**Comment by courtneyfaulkner** [comment]
Ah, interesting. I was reading your issue #4364, @cyrilgandon, and spent some time with uib 0.14.1 in plunkr. I was able to get a custom event to fire the uib-tooltip once by using `$('#tooltipTarget')[0].dispatchEvent(new Event('customEvent'));`, which is not intuitive, and also fails to work on subsequent mouseovers. Per #4364, it seems they're still trying to fix their handling of custom events. 0.14.1 seems to be handling `tooltip-is-open` functionality pretty well, but in 0.13.4, I'm seeing bad tooltip positioning problems. I updated my http://plnkr.co/edit/0wEqzz?p=preview plunkr to be similar to yours, to avoid confusion for people coming across this thread. If/when they fix custom events, I may update my plunkr again.
